src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Services/Models/TestDto.cs

```diff
@@ -39,5 +40,51 @@ public class TestDto
         public string DiscountLogoPath { get; set; }
 
         public bool HasUserAnswered { get; set; }
+
+        public static void SetIsAdvent(TestDto testDto, string defaultDateTimeFormat = "dd-MM-yyyy")
+        {
+            Microsoft.Extensions.Configuration.IConfiguration configuration = null;
+#pragma warning disable CA1062 // Validate arguments of public methods
+            var isAdventEndDate = configuration.GetSection("Competition:EndDate").Value;
+#pragma warning restore CA1062 // Validate arguments of public methods
+            var isAdventStartDate = configuration.GetSection("Competition:StartDate").Value;
+            var correctStartDateTimeFormat = DateTimeOffset.TryParseExact(isAdventStartDate, defaultDateTimeFormat,
+                CultureInfo.InvariantCulture,
+                DateTimeStyles.None, out var dateValue);
+            var correctEndDateTimeFormat = DateTimeOffset.TryParseExact(isAdventEndDate, defaultDateTimeFormat,
+                CultureInfo.InvariantCulture,
+                DateTimeStyles.None, out var dateValue1);
+
+            if (!correctStartDateTimeFormat || correctEndDateTimeFormat)
+            {
+                throw new InvalidOperationException(
+#pragma warning disable CA1303 // Do not pass literals as localized parameters
+                    $"Niepoprawny format daty zmiennej ResultPublicationDateTime w appsettings ({defaultDateTimeFormat})");
+#pragma warning restore CA1303 // Do not pass literals as localized parameters
+            }
+
+            if (Convert.ToDateTime(isAdventStartDate, CultureInfo.InvariantCulture) >= DateTime.Now
+                && Convert.ToDateTime(isAdventEndDate, CultureInfo.InvariantCulture) <= DateTime.Now)
+            {
+                if (testDto == null)
```

Comment:
this check should be at the first line of method, in 44

look on similar exception what are we setting in parameter